benches/poll.rs

```diff
@@ -0,0 +1,91 @@
+#[macro_use]
+extern crate criterion;
+
+use criterion::Criterion;
+
+use std::thread::spawn;
+use tmq::{pull, push};
+use zmq::SocketType;
+
+fn poll_benchmark(c: &mut Criterion) {
+    c.bench_function("receive", |b| {
+        let address = "tcp://127.0.0.1:3011";
+        let mut runtime = tokio::runtime::Builder::new()
+            .basic_scheduler()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        let ctx = zmq::Context::new();
+        let sender = ctx.socket(SocketType::PUSH).unwrap();
+        sender.set_linger(0).unwrap();
+        sender.connect(address).unwrap();
+
+        let ctx2 = zmq::Context::new();
+        let mut socket = runtime.enter(|| pull(&ctx2).bind(address).unwrap().unwrap());
+
+        b.iter_with_setup(
+            || {
+                sender
+                    .send_multipart(
+                        vec![zmq::Message::from("hello"), zmq::Message::from("world")].into_iter(),
```

Comment:
I tried it already before (using `impl From<Vec<Into<Message>>>`) several times, but I was surprised by some of the possible interactions:

```rust
let str = String::new();
let msg: Multipart = vec![
    str.as_bytes(),
    b"topic",
].into();
// ^ works (&[u8; 5] can be turned into &[u8])

let msg: Multipart = vec![
    b"topic",
    str.as_bytes(),
].into();
// ^ doesn't work (&[u8] can't be turned into &[u8; 5])
// you have to manually cast:
let msg: Multipart = vec![
    b"topic" as &[u8],
    str.as_bytes(),
].into();
```

It also requires to copy the vector because we need to call `into()` on each element. This is probably not a big deal as I don't expect multipart to contain a large number of messages.

I didn't find any situations where it would be a problem and we also implement `FromIterator<Message>`, so it should be fine. I added it, even though I suspect that this impl will only be useful for "Hello world" scenarios.

Reply:
Ah, now I remember why I didn't implement it, because this doesn't work:

```rust
impl <T: Into<Message>> From<Vec<T>> for Multipart {}
impl <T: Into<Message>> From<T> for Multipart {} // conflict
```
Either one of them is fine though. So we should decide what is more useful.